### PR TITLE
Jetpack Plans: Fix duplicate tracks events on plugin auto-installation

### PR DESCRIPTION
**Pull request by ryelle**
It's possible that the page state could update after the install is finished (once the sites list finally loads, for example), and this was causing the page to re-render and send the tracks events multiple times. I've added a helper functions `sendTracks`, which will ensure we only send the result once.

While debugging this, I also noticed we flash the "You need to select a jetpack site to be able to setup your plan" error if the site isn't loaded. Some of the logic here wasn't updated when we switched back to [the `sitesFactory` method of getting the selected site](https://github.com/Automattic/wp-calypso/pull/5907/commits/c244a8c4f8ad74cc3136699de857f620cbd5ccf1). We should now be checking `sites.fetching` rather than `props.isRequestingSites`.

To test:

1. Clear localstorage so the site needs to be re-fetched
2. Turn on analytics debug messages: `localStorage.setItem('debug', 'calypso:analytics');`
3. Load the autoconfig page: `/plugins/setup/$site`
4. Watch for the tracks events, there should just be one success or error message.

cc @johnHackworth @roccotripaldi 

Test live: https://calypso.live/?branch=fix/autoconfig-duplicate-tracks

### CODE REVIEW

**Comment by roccotripaldi** [comment]
Looks good. Nice cleanup. I noticed proper placeholders while sites are loading. And I'm only seeing one tracks event. We'll I'm seeing three debug messages:
screen shot 2016-07-11 at 8 39 48 pm

But I'm assuming those 3 messages correspond with 1 tracks event. I think we can :shipit:

**Comment by roccotripaldi** [comment]
Maybe we should have click events on "Contact Support" and "Manual Installation"
It's minor, and can be handled in an other PR.

**Comment by johnHackworth** [comment]
it looks that works fine for me:
image

File: client/my-sites/plugins/jetpack-plugins-setup/index.jsx [hunk 1]
```diff
@@ -58,6 +57,14 @@ const helpLinks = {
 
 const PlansSetup = React.createClass( {
 	displayName: 'PlanSetup',
+	sentTracks: false,
+
+	sendTracks( eventName, options = null ) {
```
Comment (johnHackworth) [inline client/my-sites/plugins/jetpack-plugins-setup/index.jsx]: Can we change this method name for something less generic? something like "trackConfigResults" or something in that line? With "sendTracks" it looks if it were a wrapper over the generic analytics module used in this file to register tracks events, and it's not always the case (https://github.com/Automattic/wp-calypso/pull/6690/files#diff-dd68b4cc9f9bd88ff8d710484a7b86e6R148)

**Comment by johnHackworth** [comment]
I leave a very minor comment, but let's ship this and (if you agree with my name-changing point) address it in other PR if you want! :+1:

**Pull request merged**